### DISCUSSION TITLE
Add preset navigation

### DISCUSSION THREAD

**Discussion by conversy** [christofmuc/KnobKraft-orm#209] [Ideas]
What about a small interface to navigate through presets on the synth?
Something that would select a bank and a preset with Bank Select and Program Change, together with a small name query if available in the adaptation.

That would ease getting stuff from synths that only supports sending the Edit Buffer and not Random Access, like the Solaris.

Just an idea, it's not a priority.

**Comment by christofmuc** [comment]
Oh, the Librarian will already do this for synths that have only edit buffer support - it will us the edit buffer functions and program changes, not sure about bank changes though. Which synth are you thinking about?

**Reply by conversy** [reply]
Solaris ;-) Only Edit Buffer so far (this may change in the future), but multiple banks.

**Reply by conversy** [reply]
Oh, I did not notice that v2 has this capability? (Bank select). I'll investigate further to understand how it works.

**Reply by christofmuc** [reply]
what you always can do is to simulate a programdumprequest with a bank select, program change, and then an edit buffer request - you're allowed to return multiple messages from that function in one array and they will be sent together.

**Comment by christofmuc** [comment]
@conversy Without having looked too deep into it, what you might want to implement is the "bank dump capability" - this is meant for synths that cannot send individual patches, but would send a certain number of messages for a bank. E.g. the MKS80 sends 8 messages for 64 programs, the K3 sends a single message for 64 programs but needs another call to retrieve the user wave message, and others. 

If you look at the overview table https://github.com/christofmuc/KnobKraft-orm/blob/20aea14236dca045060658023036e6392f49d90c/adaptions/implementation_overview.md it is column 13, bank dump capability. That could be used I think to retrieve the 128 edit buffers, plus the name messages, so you get all data and then need to assemble it fully before returning it as a list of patches. 

It is the 3rd capability documented in the programming guide, https://github.com/christofmuc/KnobKraft-orm/blob/master/adaptions/Adaptation%20Programming%20Guide.md.
